superset-frontend/src/explore/components/ControlPanelsContainer.tsx

```diff
@@ -272,6 +272,7 @@ export const ControlPanelsContainer = (props: ControlPanelsContainerProps) => {
   useEffect(() => {
     if (
       prevDatasource &&
+      prevDatasource.type !== DatasourceType.Query &&
```

Comment:
can we use duck typing here instead of checking the datasource type itself? Just in case some other type comes along that also meets the requirements here.